app.py

```diff
@@ -65,6 +66,54 @@
 model_data = JobShopData()
 
 
+@app.callback(
+    Output("left-column", "className"),
+    inputs=[
+        Input("left-column-collapse", "n_clicks"),
+        State("left-column", "className"),
+    ],
+    prevent_initial_call=True,
+)
+def toggle_left_column(left_column_collapse: int, class_name: str) -> str:
+    """Toggles left column 'collapsed' class that hides and shows the left column.
+
+    Args:
+        left_column_collapse (int): The (total) number of times the collapse button has been clicked.
+        class_name (str): Current class name of the left column, 'collapsed' if not visible, empty string if visible
+
+    Returns:
+        str: The new class name of the left column.
+    """
+    if class_name:
+        return ""
+    return "collapsed"
+
+
+@app.callback(
+    Output('solver-select', 'className'),
+    Output('solver-select', 'value'),
+    inputs=[
+        Input('model-select', 'value'),
+    ],
+    prevent_initial_call=True
+)
+def update_solver_options(model_value: str) -> list[str]:
+    """Hides and shows classical solver option using 'hide-classic' class
+
+    Args:
+        model_value (str): Currently selected model from model-select dropdown.
+
+    Returns:
+        str: The new class name of the solver-select checklist.
+        list: Unselects MIP and selects Hybrid or no update.
+    """
+
+    if model_value == 'QM':
```

Comment:
Slight preference to using enums for model comparisons instead of strings. Makes things a bit easier to keep track of and maintain. E.g.,

```python
from enum import Enum

class Model(Enum):
    MIP = 0
    QM = 1
```

```python
if model_value is Model.QM

Reply:
@thisac which file makes the most sense for the class to go in?

Reply:
As it's organized atm, just in `app.py` since that's the only place (I think) that it'd be used. I'd lean towards refactoring this slightly though, cleaning up the logic in the `job_shop_scheduler.py` file (specifically the `run_optimization_...` functions). It's nice to keep `app.py` as simple as possible, so you could e.g., have the `run_shop_scheduler` function take the model as an argument and put the enums in the `job_shop_scheduler.py` file.